src/key_exchange/traits.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+use crate::{
+    errors::{InternalPakeError, ProtocolError},
+    keypair::{Key, KeyPair},
+};
+use rand_core::{CryptoRng, RngCore};
+
+use std::convert::TryFrom;
+
+pub trait KeyExchange {
```

Comment:
With a monolithic trait:
- the implementer has to implement both the Client and the Server part of the protocol monolithically. As we go through the work of having a binary format for messages, it opens an opportunity for having a client / server part independent of its counterpart.
- the implementer has to implement all messages in one fell swoop. Yet e.g. a protocol without the KE3 message (i.e. server and client meet on a shared-secret-derived channel rather than HMac), would kinda make sense.

Reply:
Created https://github.com/novifinancial/opaque-ke/issues/23 to track this, but let's push this first since this change might not be the highest priority at the moment.